numpy/ma/core.py

```diff
@@ -5079,24 +5007,66 @@ def mean(self, axis=None, dtype=None, out=None):
         1.5
 
         """
+        kwargs = {}
+        if keepdims is not np._NoValue:
+            kwargs['keepdims'] = keepdims
+
         if self._mask is nomask:
-            result = super(MaskedArray, self).mean(axis=axis, dtype=dtype)
-        else:
-            dsum = self.sum(axis=axis, dtype=dtype)
-            cnt = self.count(axis=axis)
-            if cnt.shape == () and (cnt == 0):
-                result = masked
+            ret = self._data.mean(axis, dtype, out, **kwargs)
```

Comment:
I wonder if either of two large simplifications would be worth it:
1. Given your worry about inconsistencies between different versions of `mean`, and my sense that we cannot solve these anyway, could one just stick to the `super` call?
2. Since `self.count` is very quick for `nomask`, can we just remove the special path and deal with everything the same way?

I think I'd prefer (2), as it ensures that one keeps a `MaskedArray` as output (which would seem more logical).